### PR TITLE
Fix feature flags default rollout

### LINKED ISSUE #3848: Feature flags with neither rollout % nor filters say they match all users but match none
## Bug description

This is what a default feature flag looks like:
empty feature flag
It has no rollout percentage specified, nor filters, so it says "Rolling out to 100% of users". However I've experienced that in fact in this situation the flag is rolled out to *nobody*.

CC @paolodamico 

## Expected behavior

Flag should work the way it says.

## How to reproduce

1. Create a feature flag and leave it as is, supposedly completely rolled out.
2. Try to use it.
3. Discover that the flag is not rolled out.

## Environment

- [x] PostHog Cloud
- [x] self-hosted PostHog, version/commit: master (8f4ae2b)

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

Before #3675 it was quite ambiguous what happened when you created a feature flag with default params (is it released for everyone, for anyone?, ...). #3675 clarified in the UI that if you don't add any filters to a feature flag match group nor a rollout percentage, the flag is released for everyone. This however wasn't the behavior of the `/decide` endpoint.

This PR then:
- Closes #3848
- Updates the `/decide` endpoint so that a feature flag will match everyone if it has an empty group with no rollout percentage and no filters at all.
- Clarify this change in the feature flags table list UI.
- To avoid accidentally releasing feature flags when creating them, while feature flags will be matching 100% of users by default, feature flags will be disabled by default until a user deliberately enables them.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

**Comment by paolodamico** [comment]
@timgl you have a lot more context than me, so some things to keep in mind:
- Want to make sure we're not causing any backwards breaking changes.
- Looks like we could deprecate the `rollout_percentage` attribute on the Feature Flag model? As we're now setting it in `filters.groups`?
- Related to the above, I think https://github.com/PostHog/posthog/blob/a5be4f2c33a30c837d5eede883361684888d74e5/posthog/models/feature_flag.py#L58 fallback is no longer necessary either?

File: posthog/models/feature_flag.py [hunk 1]
```diff
@@ -84,11 +84,10 @@ def is_group_match(self, group: Dict, group_index: int):
...
                 return True
 
-        if rollout_percentage is not None:
-            if self._hash <= (rollout_percentage / 100):
-                return True
+        if rollout_percentage is not None and self._hash > (rollout_percentage / 100):
+            return False
 
-        return False
+        return True
```
Comment (EDsCODE) [inline posthog/models/feature_flag.py:90]: If we set the rollout_percentage by default we won't have to change this logic

**Comment by paolodamico** [comment]
Ok so tried exploring what you suggested which I think it's a pretty valid argument (see e19e97b7e30777fda0a6cedcff8611bea8375eb2). With this commit basically I would default to releasing to 100% users with `rollout_percentage`. Then you have the problem of what to do when users uncheck the rollout percentage checkbox, it no longer makes sense because if you uncheck it, it defaults to 0% if you have no filters whereas it defaults to 100% if you have filters (which is quite confusing). So it made sense to get rid of the checkbox, but this I think also introduces confusion, even more when you have filters (because now you will see 100% of users with filters which is a bit of a "make me think" problem). This particularly for existing feature flags presents an even bigger problem, because you have `rollout_percentage = null` which now doesn't make sense on the frontend.

After more thought, I think the original solution makes the most sense from a pure UX standpoint, it's what's more intuitive IMO. I'm definitely a bit hesitant because it indeed changes the default behavior for feature flags so open to more feedback before moving forward. Thoughts @EDsCODE ?

**Review by EDsCODE — APPROVED** [review]
Yeah it wasn't the biggest blocker imo. I thought it might make the numbers more consistent but it seems like it will cause a different line of UX weirdness anyway

**Pull request merged**